contents/docs/integrate/_snippets/install-vue.mdx

```diff
@@ -0,0 +1,371 @@
+import { provide, inject } from 'vue'
+import posthog from 'posthog-js'
+
+const PostHogSymbol = Symbol('PostHog')
+
+export function usePostHogProvider() {
+  const posthogInstance = posthog.init(
+    "<ph_project_api_key>",
+    {
+      api_host: "<ph_client_api_host>",
+      person_profiles: 'identified_only'
+    }
+  )
+
+  provide(PostHogSymbol, posthogInstance)
+
+  return posthogInstance
+}
+
+export function usePostHog() {
+  const posthogInstance = inject(PostHogSymbol)
+  if (!posthogInstance) {
+    throw new Error('PostHog has not been provided')
+  }
+  return posthogInstance
+}
+```
+
+In `App.vue`, call `usePostHogProvider()` to initialize PostHog.
+
+```js
+// ...rest of your app
+</template>
+
+<script setup>
+import { usePostHogProvider } from './composables/usePostHog'
+
+// Initialize and provide PostHog
+usePostHogProvider()
+</script>
+
+<script>
+// ...rest of your app
+```
+
+You can then use `usePostHog()` to access PostHog in any component.
+
+```js
+<template>
+  <div>
+    <button @click="trackEvent">Track Event</button>
+  </div>
+</template>
+
+<script setup>
+import { usePostHog } from '../composables/usePostHog'
+
+const posthog = usePostHog()
+
+const trackEvent = () => {
+  posthog.capture('button_clicked', { property: 'value' })
+}
+</script>
+```
+
+## Capturing pageviews
+
+You might notice that moving between pages only captures a single pageview event. This is because PostHog only captures pageview events when a [page load](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) is fired. Since Vue creates a single-page app, this only happens once, and the Vue router handles subsequent page changes.
+
+If we want to capture every route change, we must write code to capture pageviews that integrates with the router.
+
+First, make sure to set `capture_pageview` in the PostHog initialization config to `false`. This turns off autocaptured pageviews and ensures you won't double-capture pageviews on the first load.
+
+```js
+// in the file where you initialize posthog
+posthog.init(
+      "<ph_project_api_key>",
+      {
+        api_host: "<ph_client_api_host>",
+        person_profiles: 'identified_only',
+        capture_pageview: false
+      }
+);
+```
+
+**Vue 3.x:** 
+
+In `main.js`, set up PostHog to capture pageviews in the `router.afterEach` function. Additionally, you can use `nextTick` so that the capture event fires only after the page is mounted.
+
+```js
+// main.js
+import { createApp, nextTick } from 'vue'
+import App from './App.vue'
+import router from './router'
+import posthogPlugin from '../plugins/posthog';
+
+const app = createApp(App);
+app.use(posthogPlugin)
+.use(router)
+.mount('#app');
+
+router.afterEach((to, from, failure) => {
+  if (!failure) {
+    nextTick(() => {
+      app.config.globalProperties.$posthog.capture('$pageview', { path: to.fullPath });
+    });
+  }
+});
+```
+
+**Vue 2.x:** 
+
+```js
+export default new Router({
+	/*
```

Comment:
I might be missing something, but we might be able to remove these empty comments?

Reply:
I think it represents other code, but I will fix in the refactor of the Vue install docs.